browser/playlist/playlist_tab_helper.cc

```diff
@@ -250,9 +249,21 @@ void PlaylistTabHelper::OnFoundMediaFromContents(
 
   DVLOG(2) << __FUNCTION__ << " item count : " << items.size();
 
-  found_items_.insert(found_items_.end(),
-                      std::make_move_iterator(items.begin()),
-                      std::make_move_iterator(items.end()));
+  base::flat_map<std::string, mojom::PlaylistItemPtr*> already_found_items;
+  for (auto& item : found_items_) {
+    already_found_items.insert({item->media_source.spec(), &item});
+  }
+
+  for (auto& new_item : items) {
+    const auto media_source = new_item->media_source.spec();
+    if (base::Contains(already_found_items, media_source)) {
+      DVLOG(2) << "The media source with url (" << media_source
+               << ") already exists so update the data";
+      (*already_found_items.at(media_source)) = std::move(new_item);
+    } else {
+      found_items_.push_back(std::move(new_item));
+    }
+  }
 
   for (auto& observer : observers_) {
```

Comment:
Just curious - do we need to check `found_times_` value changes before notifying here?
Not sure it could be same but it looks like this notifying is fired always.

Reply:
Yeah, I think it'd be better to notify every time, assuming that found items' attributes like title or author could be updated.